src/workerd/io/io-context.c++

```diff
@@ -552,6 +554,32 @@ class IoContext::PendingEvent: public kj::Refcounted {
 };
 
 IoContext::~IoContext() noexcept(false) {
+  // Dispatch unload events if any were attached
+  KJ_IF_SOME(onUnload, onUnload) {
+    runInContextScope(
+        Worker::Lock::TakeSynchronously(kj::none), kj::none, [&](Worker::Lock& workerLock) {
+      jsg::Lock& js = workerLock;
+      // Run onUnload tasks without an IoContext
+      {
+        SuppressIoContextScope suppressIo;
+        js.tryCatch([&] { onUnload(workerLock); }, [&](jsg::Value exception) {
+          // Log the exception
+          js.reportError(jsg::JsValue(exception.getHandle(js)));
+        });
+      }
+
+      // Clear any microtasks that were queued during the unload handler.
+      // terminateNextExecution() causes runMicrotasks() to immediately dequeue and cancel
+      // all pending microtasks without executing them.
+      js.terminateNextExecution();
+      v8::TryCatch tryCatch(workerLock.getIsolate());
+      js.runMicrotasks();
+      if (tryCatch.HasCaught()) {
+        tryCatch.Reset();
```

Comment:
A comment here that explains the error handling a bit would be helpful for folks coming into this code later.

Reply:
Sure, added.